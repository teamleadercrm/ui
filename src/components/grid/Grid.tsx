import React from 'react';

const Grid = () => {
  return <div>Grid</div>;
};

export default Grid;
