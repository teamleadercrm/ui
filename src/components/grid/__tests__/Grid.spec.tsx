import React from 'react';
import { render } from '@testing-library/react';
import Grid, { GridItem } from '..';

describe('Component - Grid', () => {
  it('renders', async () => {
    const { asFragment } = render(
      <Grid
        areas={['header header', 'sidebar content', 'footer footer']}
        rows={['100px', '1fr', '50px']}
        columns={['20%', '1fr']}
        gap={2}
      >
        <GridItem area="header">Header</GridItem>
        <GridItem area="sidebar">Sidebar</GridItem>
        <GridItem area="content">Content</GridItem>
        <GridItem area="footer">Footer</GridItem>
      </Grid>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
