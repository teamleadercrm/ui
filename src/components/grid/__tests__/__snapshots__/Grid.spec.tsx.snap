// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Grid renders 1`] = `
<DocumentFragment>
  <div
    class="grid gap-2"
    style="grid-template-areas: "header header" "sidebar content" "footer footer"; grid-template-rows: 100px 1fr 50px; grid-template-columns: 20% 1fr;"
  >
    <div
      style="grid-area: header;"
    >
      Header
    </div>
    <div
      style="grid-area: sidebar;"
    >
      Sidebar
    </div>
    <div
      style="grid-area: content;"
    >
      Content
    </div>
    <div
      style="grid-area: footer;"
    >
      Footer
    </div>
  </div>
</DocumentFragment>
`;
